decidim-core/app/views/decidim/widgets/show.js.erb

```diff
@@ -0,0 +1,3 @@
+document.write('<link type="text/css" rel="stylesheet" media="all" href="<%= asset_url 'decidim/widget.css' %>"></link>');
```

Comment:
That's exactly it - I would reuse most of what we've done but make it write an `iframe` on the DOM instead. This way, if we ever change our mind, we keep control of the JS code. If we provided just an `iframe` it would be quite difficult to add other improvements.